lib/charms/mysql/v0/mysql.py

```diff
@@ -979,6 +979,30 @@ def grant_privileges_to_user(
             logger.warning(f"Failed to grant privileges to user {username}@{hostname}", exc_info=e)
             raise MySQLGrantPrivilegesToUserError(e.message)
 
+    def update_user_password(
+        self, username: str, new_password: str, current_server_config_password: str
```

Comment:
you should remove the current password setting here (see comment on `charm.py`)

Reply:
But I don't see why you would need to refresh it, i.e.: 

`actions is called -> new pass to mysql using old pass -> new pass to databag -> hook finish`

next runs of anything will run with new password already

Reply:
I am talking about the scenario where 
- we rotate the serverconfig password and then after that 
- try to rotate any other password. 
 
In this scenario, the update_user_password() function which relies on the serverconfig password will fail since it's copy of the serverconfig password loaded at init would be out of sync.

Reply:
ack, will update the lib's copy of serverconfig password as well to keep it in sync